lab.R

```diff
@@ -139,22 +147,23 @@ levels(bad_factor)
 
 #' To fix this, we need to set the order of the levels manually.  First, edit the next line, so that the levels correspond to the list in the documentation, with "Very Poor" as the lowest (first) value and "Very Excellent" as the highest (<http://jse.amstat.org/v19n3/decock/DataDocumentation.txt>, search for "Overall Cond").  
 
-condition_levels = c('Above_Average', 'Very_Poor', 'Below_Average')
+condition_levels = c('Very_Poor', 'Poor', 'Fair', 'Below_Average', 'Average', 'Above_Average', 'Good', 'Very_Good' , 'Excellent', 'Very_Excellent')
 
 #' 4. To fix the levels, we use the `levels = ` argument in `factor()`.  Modify the code for `bad_factor` above, assigning the result to `good_factor`.  Confirm that the levels are in the right order. 
 
-# good_factor = ???
+good_factor = factor(dataf$Overall_Cond, levels= condition_levels)
+levels(good_factor)
 
 #' 5. Finally we want this factor to be in our analysis dataframe.  *Normally, to preserve immutability*, I would do this right after reading in the CSV (using a pipe).  We'll talk more about this when we work on cleaning data.  For this lab, just do it here.  
 #' 
 #' Using `mutate()`, add a variable `overall_cond_fct` to `dataf`, that represents overall condition as a factor with the levels in the correct order. 
 
-# dataf = ???
+dataf = mutate(dataf,overall_cond_fact = good_factor)
 
 
 #' # Problem 6 #
 #' Recall that we're interested in finding variables that are highly correlated with sale price.  We can use the function `cor()` to construct a correlation matrix, with correlations between all pairs of variables in the dataframe.  But this creates two challenges.  First, `cor()` only works with numerical inputs.  If we try it with our current dataframe, it throws an error:  
-# cor(dataf)
+cor(dataf)
```

Comment:
This is here so you can see that it throws an error.  But we don't want to throw the error. 
```suggestion
# cor(dataf)
```

Reply:
My mistake, I uncommented it to see to output and forgot to comment it out back.